sweep/tx_input_set.go

```diff
@@ -179,11 +188,28 @@ func (t *txInputSet) addToState(inp input.Input,
 
 	// If the input comes with a required tx out that is below dust, we
 	// won't add it.
+	//
+	// NOTE: only HtlcSecondLevelAnchorInput returns non-nil RequiredTxOut.
 	reqOut := inp.RequiredTxOut()
 	if reqOut != nil {
 		// Fetch the dust limit for this output.
+		//
+		// TODO(yy): replace this check with `mempool.IsDust` to take
+		// the minRelayTxFee into account in case it's not using the
+		// default value.
 		dustLimit := lnwallet.DustLimitForSize(len(reqOut.PkScript))
 		if btcutil.Amount(reqOut.Value) < dustLimit {
+			log.Debugf("Reject input=%v due to dust required "+
+				"output=%v, limit=%v", inp, reqOut.Value,
+				dustLimit)
+
+			// TODO(yy): we should not return here for force
```

Comment:
yeah imo we should return an error here - so that the caller know this would fail.

Reply:
> But I think we cannot just continue for Force-Sweeps as well if we hit this case, because the input is signed with the SIGHASH-Single|AnyoneCanPay

What do you mean?

> In case we remain with the default mempool feelimit of bitcoind (3000 sats/vbyte) I think this case should be marked as an error

Yeah agree, changed to log an error instead.

Reply:
Ok I was not 100% in regards of the comment, why do you want to distinguish here for force and non-forced sweeps  and why should we not return early if we figure out that this is a required dust-output ?

Reply:
It really depends on how we define `force`. Atm it means to collect the input although it may be uneconomical, while I think what it really means is, this output needs to be swept because it has other implications, such as CPFP the closing tx, and we should make it happen and fail loudly when it fails.